core/src/main/java/com/google/instrumentation/trace/TraceService.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2016, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.trace;
+
+/**
+ * Class that holds the implementation instances for {@link Tracer} and {@link
+ * BinaryPropagationHandler}.
+ *
+ * <p>This implementation of this class is loaded using reflection by the {@link Trace}.
+ */
+public abstract class TraceService {
```

Comment:
if this is only used for testing, I'd be inclined to inline its code back into Trace.

you can use Powermock to mock static methods. That eliminates the need to explain the otherwise public supporting type here. Powermock can do fancy things as it often actually wraps the callsite as opposed to changing the target. We use it in zipkin sometimes.. I like it a lot better than putting test code in main, but it is a judgement call.

ex.

```java
@RunWith(PowerMockRunner.class)
@PrepareForTest({MyCode.class}) // implicitly intercepts the call site
public class PlatformTest {
  MyCode code = new CodeThatUsesTraceDotGetTracer();
  Tracer tracer = newFooTracer();

  @Before public void setup() {
    mockStatic(Trace.class);
    when(Trace.getTracer()).thenReturn(mockTracer);
   }
```

```xml
    <dependency>
      <groupId>org.powermock</groupId>
      <artifactId>powermock-api-mockito</artifactId>
      <version>${powermock.version}</version>
      <scope>test</scope>
    </dependency>
```